src/environmentbase/networkbase.py

```diff
@@ -347,11 +373,62 @@ def add_vpn_gateway(self,
         else:
             vpn_name = self.__class__.__name__ + 'Gateway'
 
-        gateway = self.template.add_resource(ec2.VPNGateway('vpnGateway',
+        gateway = self.add_resource(ec2.VPNGateway('vpnGateway',
             Type=vpn_conf.get('vpn_type', 'ipsec.1'),
             Tags=[ec2.Tag(key='Name', value=vpn_name)]))
 
-        gateway_connection = self.template.add_resource(ec2.VPCGatewayAttachment('vpnGatewayAttachment',
-            VpcId=self.template.vpc_id,
-            InternetGatewayId=self.template.igw,
+        gateway_connection = self.add_resource(ec2.VPCGatewayAttachment('vpnGatewayAttachment',
+            VpcId=self.vpc_id,
+            InternetGatewayId=self.igw,
             VpnGatewayId=gateway))
+
+
+class NetworkBase(EnvironmentBase):
+    """
+    EnvironmentBase controller containing a root template with all of the base networking infrastructure
+    for a common deployment within AWS. This is intended to be the 'base' stack for deploying child stacks
+    """
+
+    def create_hook(self):
+        network_config = self.config.get('network', {})
+        boto_config = self.config.get('boto', {})
+        nat_config = self.config.get('nat')
+        az_count = int(network_config.get('az_count', '2'))
+
+        base_network_template = BaseNetwork('BaseNetwork', network_config, boto_config, nat_config, az_count)
+        self.add_child_template(base_network_template)
+
+        # TODO: make outputs passing less fragile:
+        # base_network_template.build_hook()
+        # for output in base_network_template.outputs:
+        #     self.template.add_output(Output('', Value=GetAtt(base_network_template.name, '')))
+
+        self.template._subnets = base_network_template._subnets
+
+        self.template.add_output(Output('commonSecurityGroupId', 
+                Value=GetAtt(base_network_template.name, 'Outputs.commonSecurityGroupId')))
+
+        self.template.add_output(Output('vpcId', 
+                Value=GetAtt(base_network_template.name, 'Outputs.vpcId')))
+        self.template._vpc_id = GetAtt(base_network_template.name, 'Outputs.vpcId')
```

Comment:
@gimballock we need to do this for the other python attributes, `._vpc_cidr`, ...